### PR TITLE
mayor driver sdk refactor

### PR DESCRIPTION
**Pull request by mcuadros**
Sorry for how big is this PR, but was more or less a non-stop working weekend. 

Improvements:
- Communication between server and driver based on grpc, the drivers now use the `protocol` package and they are able to server a grpc server via tcp, for testing and unix sockets for server (configurable via cli)
- Package `driver` and other move outside of `protocol` since is not part of the protocol, now they are under `sdk` package.
- Manifest is read and can be access by the `driver` package. 
- Protocol: New NativeParseRequest, to be able to retrieve original AST, via gRPC
- Protocol: `VersionResponse` now it based on manifest.
- Integration testing: now we don't use anymore a bash, its based on `go test` and the test are perform via `bblfsh-sdk-tools test` or `make integration-test`, the test runs against any gRPC server implementing the protocol.

### CODE REVIEW

File: sdk/driver/native.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"gopkg.in/bblfsh/sdk.v1/protocol"
+	"gopkg.in/bblfsh/sdk.v1/sdk/jsonlines"
+
+	"srcd.works/go-errors.v0"
```
Comment (juanjux) [inline sdk/driver/native.go]: gopkg.in/src-d/go-errors.v1?

File: sdk/server/server.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package server
+
+import (
+	"net"
+
+	"gopkg.in/bblfsh/sdk.v1/protocol"
+
+	"github.com/Sirupsen/logrus"
+	"google.golang.org/grpc"
+	"srcd.works/go-errors.v0"
```
Comment (juanjux) [inline sdk/server/server.go]: gopkg.in/src-d/go-errors.v1?

File: etc/build/etc/it.bash [hunk 3]
```diff
@@ -1,96 +0,0 @@
-#!/bin/bash -e
```
Comment (juanjux) [inline etc/build/etc/it.bash]: :+1:

File: protocol/service_test.go [hunk 4]
```diff
@@ -191,3 +177,39 @@ func checkError(err error) {
 		panic(err)
 	}
 }
+
+// ServiceMock implements protocol.Service interface implement the methods to
```
Comment (juanjux) [inline protocol/service_test.go]: Maybe better: "ServiceMock implements the protocol.Servce interface and the methods to be used in the test"

File: cli/bblfsh-sdk-tools/cmd/manifest.go [hunk 5]
```diff
@@ -1,44 +1,27 @@
...
 
 	"gopkg.in/bblfsh/sdk.v1/cli"
 	"gopkg.in/bblfsh/sdk.v1/manifest"
 )
 
-const ManifestCommandDescription = "" +
-	"prints the manifest as a list of variables ready to be evaluated by bash or make"
-
-type ManifestCommand struct {
+type manifestCommand struct {
```
Comment (abeaumont) [inline cli/bblfsh-sdk-tools/cmd/manifest.go]: Manifest command is not a command anymore, just some helper code to read a manifest file, shouldn't it be named accordingly?

File: sdk/driver/driver.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+		r.Elapsed = time.Since(start)
+	}()
+
+	var ast interface{}
+	r.Response, ast = d.doParse(req.Language, req.Content, req.Encoding)
+	if r.Status == protocol.Fatal {
+		return r
+	}
+
+	var err error
```
Comment (juanjux) [inline sdk/driver/driver.go]: Is this needed? Shouldn't be enough with `r.UAST, err :=` below? (same for the `var ast interface{}` above and below).

File: cli/bblfsh-sdk-tools/cmd/manifest.go (hunk 5 above)
Comment (mcuadros) [reply cli/bblfsh-sdk-tools/cmd/manifest.go]: This pattern is already present at https://github.com/bblfsh/sdk/blob/master/cli/common.go. I can change both if you want, for something like `Helper`

Comment (abeaumont) [reply cli/bblfsh-sdk-tools/cmd/manifest.go]: As long as we're consistent, I'm fine with it. Whatever @juanjux prefers, I guess.

File: sdk/driver/driver_test.go [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	require.NotNil(d)
+
+	err = d.Start()
+	require.NoError(err)
+
+	r := d.Parse(&protocol.ParseRequest{
+		Language: "fixture",
+		Content:  "foo",
+	})
+
```
Comment (juanjux) [inline sdk/driver/driver_test.go]: Nitpicking: Inconsistent newline (other tests have the NoError/NotNil test just below the calls/struct literals).

File: sdk/driver/driver.go (hunk 6 above)
Comment (mcuadros) [reply sdk/driver/driver.go]: nope, because r.UAST is already defined.

**Pull request merged**